pyinfra/api/state.py

```diff
@@ -362,3 +365,43 @@ def get_temp_filename(self, hash_key=None, hash_filename=True):
             hash_key = sha1_hash(hash_key)
 
         return "{0}/pyinfra-{1}".format(self.config.TEMP_DIR, hash_key)
+
+    def connect_all(self):
```

Comment:
I think this change is sensible, we should make the original functions just call these though to avoid two implementations that may diverge.